darts/models/forecasting/regression_ensemble_model.py

```diff
@@ -112,23 +131,77 @@ def _split_multi_ts_sequence(
     def fit(
         self,
         series: Union[TimeSeries, Sequence[TimeSeries]],
-        past_covariates: Optional[Union[TimeSeries, Sequence[TimeSeries]]] = None,
-        future_covariates: Optional[Union[TimeSeries, Sequence[TimeSeries]]] = None,
+        past_covariates: Optional[Union[TimeSeries, Sequence[TimeSeries], None]] = None,
+        future_covariates: Optional[
+            Union[TimeSeries, Sequence[TimeSeries], None]
+        ] = None,
     ):
+        """
+        Fits the forecasting models with the entire series except the last `regression_train_n_points` values, which
+        are used to train the regressor model.
+
+        If `forecasting_models` contains fitted `GlobalForecastingModels` and `retrain_forecasting_model=False`, the
+        regression model will be trained.
+
+        Parameters
+        ----------
+        series
+            TimeSeries or Sequence[TimeSeries] object containing the target values.
+        past_covariates
+            Optionally, a series or sequence of series specifying past-observed covariates passed to the
+            forecasting models
+        future_covariates
+            Optionally, a series or sequence of series specifying future-known covariates passed to the
+            forecasting models
+        """
 
         super().fit(
             series, past_covariates=past_covariates, future_covariates=future_covariates
         )
 
         # spare train_n_points points to serve as regression target
         is_single_series = isinstance(series, TimeSeries)
-        if is_single_series:
-            train_n_points_too_big = len(self.training_series) <= self.train_n_points
-        else:
-            train_n_points_too_big = any(
-                [len(s) <= self.train_n_points for s in series]
+        if self.train_n_points == -1:
```

Comment:
can we not use the all the points of all time series instead of using the largest possible length for all of them?
The global models are able to handle multiple time series with different lengths

Reply:
Because otherwise, there will be a shift in the time index of the "`future_covariates`" generated by the `forecasting models` and the target of the regressor: covariates will be shorter than the targets because of the lags/input_chunk_length